streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -384,6 +413,23 @@ public void punctuate(long streamTime) {
     }
 
     /**
+     * A processor that forwards messages with modified values (without custom timestamp information) to each child, if
+     * the value is in ".*@[0-9]+" format.
+     */
+    protected static class ValueTimestampProcessor extends AbstractProcessor<String, String> {
```

Comment:
We want to test that the extracted timestamp is forwarded with the record produced. We need to remove the custom timestamp information from the value, otherwise, it will be extracted from the value again.